### PR TITLE
feat: allow for configuring of revisionHistoryLimit

### LINKED ISSUE #24: RevisionHistoryLimit should be configurable
RevisionHistoryLimit can be useful to change from the default of 10 for UX reasons. Being able to configure that inside the chart would be helpful

### PR DESCRIPTION
**Pull request by knechtionscoding**
Closes #24

Chose to put it in an if clause to prevent unintended side effects; while allowing those who want to to set the value properly.

### CODE REVIEW

File: charts/pact-broker/values.yaml [hunk 1]
```diff
@@ -486,6 +489,8 @@ serviceAccount:
   # If not set and `serviceAccount.create` is true, a name is generated
   name: broker-sa
 
+  customSA
```
Comment (ChrisJBurns) [inline charts/pact-broker/values.yaml]: not sure if this has been left over?

**Comment by ChrisJBurns** [comment]
Have you run the pre-commit config? In order to generate the Helm docs for the additional value fields?

**Comment by knechtionscoding** [comment]
> Have you run the pre-commit config? In order to generate the Helm docs for the additional value fields?

Done!

File: charts/pact-broker/values.yaml [hunk 2]
```diff
@@ -35,8 +35,11 @@ broker:
   # -- Additional annotations that can be added to the Broker deployment
   annotations: {}
 
-  # -- Number of Pact Broker replicas to deploy
-  replicaCount: 1
+  # -- (int) Number of Pact Broker replicas to deploy
+  replicaCount:
```
Comment (ChrisJBurns) [inline charts/pact-broker/values.yaml]: Not should why we are removing the default value here of 1?

Comment (knechtionscoding) [reply charts/pact-broker/values.yaml]: Typo, apologies, was going fast last night

**Comment by ChrisJBurns** [comment]
@KnechtionsCoding  I think the version just needs a bump. As this is a backwards compatible feature, it will be a minor, so `0.5.0` :+1: 

I'll be putting up a contributing guide as well just so folks are aware of what they need to change. Just do a search for `0.4.2` and replace with `0.5.0`. I think you've done it for the README badges, but the main `Chart.yaml` will need it also.

**Comment by ChrisJBurns** [comment]
@KnechtionsCoding  Thanks for this. I'm going to get https://github.com/pact-foundation/pact-broker-chart/pull/29 in first, I can then use this PR in order to test the new release note generation that PR#29 implements. So merge `master` back into your branch and we can get this in :+1:

**Comment by knechtionscoding** [comment]
@ChrisJBurns merged trunk back; should be ready to go for you.

**Review by ChrisJBurns — APPROVED** [review]

**Pull request merged**